handle semodule version >=2.4 (#37732) and fix typo

dhaines opened the pull request:
### What does this PR do?
Checks for more recent versions of semodule (v2.4+) and runs semodule differently depending on what side of the version line it's on, storing different results in the dictionary

### What issues does this PR fix or reference?
#37732

### Previous Behavior
errors on any state.apply of a selinux.module state

### New Behavior
no more errors, but no more selinux module versioning for newer versions of the selinux userspace (but that's upstream's choice)

### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

Commit:
handle semodule version >=2.4 (#37732) and fix typo

Commit:
fix typo

missed a pair of parens

Commit:
fix indexing error

both the existing code and my addition erroneously strip the first result from the semodule call

cachedout commented on salt/modules/selinux.py:
```diff
@@ -240,16 +240,36 @@ def list_semod():
-                             'Version': comps[1]}
+    semodule_versioncheck = __salt__['cmd.run']('semodule -h').splitlines()
+    semodule_version = ''
+    for line in versioncheck[1:]:
```
`semodule_versioncheck`? This looks like it's undefined.

cachedout commented on salt/modules/selinux.py:
```diff
@@ -240,16 +240,36 @@ def list_semod():
+    if semodule_version == 'new':
+        mdata = __salt__['cmd.run']('semodule -lfull').splitlines()
+        ret = {}
+        for line in mdata[0:]:
```
Is there a reasoning for splicing this as you've done? `for line in mdata:` should be fine.

dhaines replied:
I was merely parroting what was there before (well, it was mdata[1:] for some reason, but I digress).

cachedout commented on salt/modules/selinux.py:
```diff
@@ -240,16 +240,36 @@ def list_semod():
+    else:
+        mdata = __salt__['cmd.run']('semodule -l').splitlines()
+        ret = {}
+        for line in mdata[0:]:
```
Same question about the splice here.

Commit:
Update selinux.py

Fixed (hopefully) a couple of issues with the commit. That's what I get for parroting (and for not really knowing Python syntax yet).

dhaines commented:
Do my fixes address your concerns?

cachedout commented:
@dhaines Looks great. Thanks so much!
